### DISCUSSION TITLE
Get current state of expanded groups in slickgrid.

### DISCUSSION THREAD

**Discussion by adityasood11** [ghiscoding/Angular-Slickgrid#1254] [Q&A]
Hi,

Is there a way to get the current state of expanded groups in slickgrid?

**Comment by ghiscoding** [comment]
if you're talking about regular Grouping (not Tree Data), then I guess you could use the [`onGroupExpanded`](https://github.com/6pac/SlickGrid/blob/master/slick.dataview.js#L797) and [`onGroupCollapsed`](https://github.com/6pac/SlickGrid/blob/master/slick.dataview.js#L78-L79) events so it could be doable to track them yourself and build a state of them

Please consider asking questions on [Stack Overflow](https://stackoverflow.com/search?tab=newest&q=slickgrid) which is a better platform and has more users who can answer your question.

**Comment by adityasood11** [comment]
I am trying to implement this in tree data. I am working on a functionality which allows me to retain my grid state after the user edits values in some of the columns of the grid. 

Closing this discussion as I have opened a thread on Stack Overflow.
https://stackoverflow.com/q/77104126/9594505

Thanks!

**Reply by ghiscoding** [reply]
if you are using Tree Data, like in [Example 28](https://ghiscoding.github.io/Angular-Slickgrid/#/tree-data-parent-child) or Example 29, then you have to mention it and that is already part of the [Grid State & Presets](https://github.com/ghiscoding/angular-slickgrid/wiki/Grid-State-&-Preset). You did not mention that detail in your Stack Overflow, in fact there's barely enough details to help with that question. The events for regular Grouping and Tree Data are totally different. For Tree Data grids, you can use the `TreeDataService` multiple methods to get that info and I updated the [Tree Data - Wiki](https://github.com/ghiscoding/Angular-Slickgrid/wiki/Tree-Data-Grid#tree-data-service-methods) to show some examples
